x-pack/plugins/metrics_entities/server/services/install_mappings.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ElasticsearchClient } from 'kibana/server';
+
+import { Mappings } from '../modules/types';
+import type { Logger } from '../../../../../src/core/server';
+
+import { computeMappingId } from './utils';
+import { getIndexExists } from './utils/get_index_exists';
+import { logMappingDebug } from './utils/log_mapping_debug';
+import { logMappingError } from './utils/log_mapping_error';
+
+interface CreateMappingOptions {
+  esClient: ElasticsearchClient;
+  mappings: Mappings[];
+  prefix: string;
+  suffix: string;
+  logger: Logger;
+}
+
+export const installMappings = async ({
+  esClient,
+  mappings,
+  prefix,
+  suffix,
+  logger,
+}: CreateMappingOptions): Promise<void> => {
+  for (const mapping of mappings) {
+    const { index } = mapping.mappings._meta;
+    const mappingId = computeMappingId({ id: index, prefix, suffix });
+    const exists = await getIndexExists(esClient, mappingId);
+    const computedBody = {
+      ...mapping,
+      ...{
+        mappings: {
+          ...mapping.mappings,
+          _meta: {
+            ...mapping.mappings._meta,
+            // TODO: Do we want to keep the version as 1.0.0 here? If so then this should be moved into constants
+            ...{ created_by: 'metrics_entities', index: mappingId, version: { created: '1.0.0' } },
```

Comment:
Is the stack version being tracked as well? This is the version of the transform mapping as released, can they be modified manually by the user in-between releases?

Reply:
In the transforms, they look to automatically add the stack version. I think from discussions we need to have at the very least the stack version in both the transforms and the mappings.

As discussed, the user can actually manually modify separate transforms outside of the these and point them to mappings that have a `_1` or other suffix. That suffix can then be used if I add a `*` at the end of the transform index and we can encourage users to go that route rather than directly modify/manipulate transforms and our mappings.

Users will need to make compatible mappings though.

Reply:
Added a star pattern to the backend of the queries such as this: `.estc_all_user_ent*`, to this PR so we are good for users to use the star pattern for the end of their queries.